### DISCUSSION TITLE
No keyboard when booting Cisco UCS-X with iPXE UEFI

### DISCUSSION THREAD

**Discussion by nies-ch** [ipxe/ipxe#1246] [Q&A]
Hi,

We have Cisco UCSX-210C-M7 blades and when booting iPXE in UEFI then the keyboard does not work via Intersight KVM once iPXE loaded. I downloaded latest iPXE from github and compiled with:

make bin-x86_64-efi/ipxe.efi

I also tried compiled with enabled USB_HCD_USBIO in config/usb.h with no luck.

Looks like a driver issue on this new hardware, but I've found no fix so far.

Regards,
Bernd

**Comment by NiKiZe** [comment]
By default iPXE does not have USB drivers enabled, so all input is thru the EFI Console.

You could try a build of snp+ecm.efi to see if that makes any difference, if it does we can find a more appropriate build for actual use.
